nixos/gitea: fix pre start script

Gerschtli opened the pull request:
###### Motivation for this change

The gitea path is hardcoded in hooks directory in files of paths like:
    
    repositories/<user>/<repo>.git/hooks/update.d/gitea

The locales can be updated on a new gitea release and therefore should be updated in the `stateDir`.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

nyanloutre commented:
I had one problem, the `cp -r` created a directory at the wrong path : 
`conf/locale/locale` instead of `conf/locale`
Also the `mkdir` seems redundant as cp can create it

uvNikita commented:
I'm not sure if it's relevant to this PR, but I'm running gitea on nixos 18.03 and currently, it's failing to start with this error message:
```
panic: fail to set message file(uk-UA): open conf/locale/locale_uk-UA.ini: no such file or directory
```

nyanloutre commented:
@uvNikita yes, this PR fix exactly this. For a quick fix you can run the command manually.

Gerschtli commented:
This feature fixes exactly this error.Am 23.08.2018 10:00 schrieb Nikita Uvarov <notifications@github.com>:I'm not sure if it's relevant to this PR, but I'm running gitea on nixos 18.03 and currently, it's failing to start with this error message:
panic: fail to set message file(uk-UA): open conf/locale/locale_uk-UA.ini: no such file or directory


—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.

uvNikita commented:
Great! Thanks for the explanation, I'll try to run the command manually for now.

xeji commented:
Merge conflict because a similar fix was merged in #45608. I'm afraid this PR (which was first) was overlooked. Please close it if the issue is fixed now.

Thank you for contributing!

Gerschtli committed:
nixos/gitea: fix pre start script

The gitea path is hardcoded in hooks directory in files of paths like:
    repositories/<user>/<repo>.git/hooks/update.d/gitea

Gerschtli commented:
Ah okay, I have updated the PR accordingly.

xeji commented:
Thank you.
